projects/frontegg-app/src/lib/frontegg-app.service.ts

```diff
@@ -97,13 +97,29 @@ export class FronteggAppService {
       }
     };
 
-    const { contextOptions } = this.config ?? {};
-    contextOptions.metadataHeaders = {
+
+    const metadataHeaders: MetadataHeaders = {
       fronteggSdkVersion: `@frontegg/angular@${sdkVersion.version}`,
-      //TODO: remove this ts-ignore after updating rest-api context options type to accept string.
-      //@ts-ignore
+      // TODO: remove this any type after updating rest-api context options type to accept string.
+      // @ts-ignore
       framework: `${FronteggFrameworks.Angular}@${angularCoreVersion.version}`,
-    }
+    };
+
+    // prepare config default values
+    this.config = {
+      ...this.config,
+      contextOptions: {
+        ...this.config.contextOptions,
+        metadataHeaders,
```

Comment:
I think this will override the user metadata headers; maybe it is worth deep merging them?

Reply:
See previous flow, it was implicitly set and override with metadataHeaders.
In this PR we are fixing other things, I will open a ticket for providing option to override the metadataHeaders